packages/eslint-plugin/src/rules/no-deprecated-type-references.ts

```diff
@@ -0,0 +1,339 @@
+        "INavbarHeadingProps",
+        "INonIdealStateProps",
+        "IOverflowListProps",
+        "IOverlayableProps",
+        "IOverlayProps",
+        "IPanel",
+        "IPanelActions",
+        "IPortalProps",
+        "IProgressBarProps",
+        "IResizeSensorProps",
+        "IHandleProps",
+        "IMultiSliderProps",
+        "IRangeSliderProps",
+        "ISliderProps",
+        "ITabProps",
+        "ITabsProps",
+        "ITabTitleProps",
+        "ITagProps",
+        "ITagInputProps",
+        "ITextProps",
+        "IToastProps",
+        "ITreeProps",
+    ],
+
+    "@blueprintjs/datetime": [
+        "IDateFormatProps",
+        "IDateInputProps",
+        "IDatePickerProps",
+        "IDatePickerModifiers",
+        "IDateRangeInputProps",
+        "IDateRangeShortcut",
+        "ITimePickerProps",
+    ],
+
+    "@blueprintjs/docs-theme": ["IExampleProps", "INavMenuItemProps", "ITagRendererMap"],
+
+    "@blueprintjs/popover2": ["IPopover2Props", "IPopover2TargetProps", "IPopover2SharedProps", "ITooltip2Props"],
+
+    "@blueprintjs/select": [
+        "IItemListRendererProps",
+        "IItemModifiers",
+        "IItemRendererProps",
+        "IListItemsProps",
+        "ICreateNewItem",
+        "IMultiSelectProps",
+        "IOmnibarProps",
+        "IQueryListProps",
+        "ISelectProps",
+        "ISuggestProps",
+    ],
+
+    "@blueprintjs/table": [
+        "IStyledRegionGroup",
+        "ICellInterval",
+        "ICellCoordinate",
+        "IRegion",
+        "ITableBodyProps",
+        "ITableProps",
+        "ICellRenderer",
+        "IJSONFormatProps",
+        "ITruncatedFormatProps",
+        "IColumnHeaderCellRenderer",
+        "IColumnHeaderCellProps",
+        "IRowHeaderRenderer",
+        "IRowHeaderCellProps",
+    ],
+
+    "@blueprintjs/timezone": ["ITimezoneItem", "ITimezonePickerProps"],
+};
+
+// some extra type mappings which do not fit the simple pattern of dropping the "I" prefix,
+// these will be augmented to create the full list in the for-loop below
+const DEPRECATED_TYPES_MAPPING: Record<string, string> = {
+    IPortalContext: "PortalLegacyContext",
+    IToaster: "ToasterInstance",
+    ITreeNode: "TreeNodeInfo",
+    ContextMenu2RenderProps: "ContextMenu2ContentProps",
+};
+
+// which packages to find the new types in
+const NEW_TYPE_PACKAGE_MAPPING: Record<string, string> = {
+    PortalLegacyContext: "@blueprintjs/core",
+    ToasterInstance: "@blueprintjs/core",
+    TreeNodeInfo: "@blueprintjs/core",
+    ContextMenu2ContentProps: "@blueprintjs/popover2",
+};
+
+for (const [packageName, deprecatedTypeNames] of Object.entries(DEPRECATED_TYPE_REFERENCES_BY_PACKAGE)) {
+    for (const typeName of deprecatedTypeNames) {
+        const newTypeName = typeName.slice(1);
+        // "IProps": "Props"
+        DEPRECATED_TYPES_MAPPING[typeName] = newTypeName;
+        // "Props": "@blueprintjs/core"
+        NEW_TYPE_PACKAGE_MAPPING[newTypeName] = packageName;
+    }
+}
```

Comment:
i kind of wonder if this is logic we should be running unilaterally as part of importing this rule, or if it should instead be worked into the logic within the rule but without the for loop.

Reply:
moved it into the `create` callback